src/libfuncs/felt252_dict_entry.rs

```diff
@@ -89,99 +88,50 @@ pub fn build_get<'ctx, 'this>(
             .alloca1(context, location, key_ty, key_layout.align())?;
     entry.store(context, location, entry_key_ptr, entry_key)?;
 
-    // Runtime's dict_get returnes a pointer to the entry's value, which is a
-    // pointer itself. Effectively, we have a double pointer to the value,
-    // avoiding a second call into the runtime when finalizing the entry.
-    let entry_value_ptr_ptr = metadata
+    let (is_present, value_ptr) = metadata
         .get_mut::<RuntimeBindingsMeta>()
         .ok_or(Error::MissingMetadata)?
         .dict_get(context, helper, entry, dict_ptr, entry_key_ptr, location)?;
-    let entry_value_ptr = entry.load(
-        context,
-        location,
-        entry_value_ptr_ptr,
-        llvm::r#type::pointer(context, 0),
-    )?;
-
-    let null_ptr =
-        entry.append_op_result(llvm::zero(llvm::r#type::pointer(context, 0), location))?;
-
-    // If the entry_value_ptr is null, then the entry is vacant
-    let is_vacant = entry.append_op_result(
-        ods::llvm::icmp(
-            context,
-            IntegerType::new(context, 1).into(),
-            entry_value_ptr,
-            null_ptr,
-            IntegerAttribute::new(IntegerType::new(context, 64).into(), 0).into(),
-            location,
-        )
-        .into(),
-    )?;
+    let is_present = entry.trunci(is_present, IntegerType::new(context, 1).into(), location)?;
 
     let value = entry.append_op_result(scf::r#if(
-        is_vacant,
+        is_present,
         &[value_ty],
         {
-            // If the entry is vacant, then create the default value
```

Comment:
Could you re-add some of the removed comments? I think they are useful for newcomers to the code.